src/inverted-index.js

```diff
@@ -0,0 +1,200 @@
+"use strict";
+
+const hasProperty = Object.prototype.hasOwnProperty;
+
+class Index {
+  constructor() {
+    // an empty object that keeps track of files uploaded
+    // and their index
+    this.filesIndexed = {};
+  }
+  
+
+  /**
+    createIndex function is used to get all the index
+    @param {object} jsonData- the json data to index
+    @parame {string} filename- the name of the file to be indexed
+    @return {boolean} true or false if the createIndex was successful
+  */
+  createIndex(jsonData, filename) {
+    //checks if the jsondata is accurate and not empty
+    try {
+      jsonData = JSON.parse(jsonData);
+    } catch(e) {
+      return false;
+    }
+
+    //adds the filename as a key to the filesIndex and
+    //initializes it to an empty object
+    this.filesIndexed[filename] = {};
+    if(!this.prepareJsonData(jsonData, filename)) {
+      //if the file was not indexed, the key is removed
+      delete this.filesIndexed[filename];
+      return false;
+    }
+    return true;
+
+  }
+
+  /**
+  prepareJsonData gets the json ready for indexing by tokenizing the
```

Comment:
**NITPICK** -- Comments should always be started with uppercase. I don't follow it sometimes